internal/authentication/azure/metadata-properties.go

```diff
@@ -35,6 +35,9 @@ var MetadataKeys = map[string][]string{ //nolint:gochecknoglobals
 	// Identifier for the Azure environment
 	// Allowed values (case-insensitive): AzurePublicCloud/AzurePublic, AzureChinaCloud/AzureChina, AzureUSGovernmentCloud/AzureUSGovernment
 	"AzureEnvironment": {"azureEnvironment", "azureCloud"},
+	// Identifier for the Azure authentication methods to try (in order), comma-separated
+	// Allowed values (case-insensitive): ServicePrincipal, Certificate, WorkloadIdentity, ManagedIdentity, CLI, None
+	"AzureAuthMethods": {"azureAuthMethods", "azureADAuthMethods"},
```

Comment:
Should it be "azureADAuthMethods" as default maybe?

Reply:
No because of the entra ID nonsense. That's why I chose this name. Please see what I had written on the linked issue.

Reply:
Ok. How about you already pre-add "entraIDAuthMethods" and "azureEntraIDAuthMethods" as aliases then?

Reply:
It's called MicrosoftEntraID - not AzureEntraID 😄

Reply:
That's why I chose "azureAuthMethods" as the default - because it is neither about Azure AD, nor about Microsoft Entra ID :)

Reply:
Damn marketers :D